### PR TITLE
feat: enable eigen matrix inversion

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR enables eigen matrix inversion, also optimizes eigen matrix by using faster constants.

### CODE REVIEW

File: tachyon/math/finite_fields/generator/generator_util.cc [hunk 1]
```diff
@@ -18,14 +20,42 @@ namespace tachyon::math {
...
-  std::vector<std::string> ret;
-  ret.reserve(limb_size);
+  enum class Type {
+    kMaxIsInt,
+    kMaxIsUint32,
+    kMaxIsUint64,
+  };
+
+  Type type = Type::kMaxIsInt;
   for (size_t i = 0; i < limb_size; ++i) {
```
Comment (TomTaehoonKim) [inline tachyon/math/finite_fields/generator/generator_util.cc:33]: I

File: tachyon/math/finite_fields/generator/generator_util.cc [hunk 2]
```diff
@@ -18,14 +20,42 @@ namespace tachyon::math {
...
-    ret.push_back(absl::Substitute("UINT64_C($0)", gmp::GetLimb(m, i)));
+    mp_limb_t limb = gmp::GetLimb(m, i);
```
Comment (TomTaehoonKim) [inline tachyon/math/finite_fields/generator/generator_util.cc:34]: I'm not sure what you are trying to do here. Shouldn't you have a vector of Type?

File: tachyon/c/math/elliptic_curves/generator/ext_field.h.tpl [hunk 3]
```diff
@@ -46,6 +46,13 @@ TACHYON_C_EXPORT tachyon_%{type}_fq%{degree} tachyon_%{type}_fq%{degree}_zero();
  */
 TACHYON_C_EXPORT tachyon_%{type}_fq%{degree} tachyon_%{type}_fq%{degree}_one();
 
+/**
+ * @brief Initializes an Fq%{degree} element to minus one.
```
Comment (TomTaehoonKim) [inline tachyon/c/math/elliptic_curves/generator/ext_field.h.tpl:50]: I'm not sure "Initializes" is the best here. Maybe returns?

File: tachyon/math/finite_fields/generator/generator_util.cc (hunk 2 above)
Comment (chokobole) [reply tachyon/math/finite_fields/generator/generator_util.cc:34]: What this code does is figure out the maximum type in the [std::initializer_list<T>](https://en.cppreference.com/w/cpp/utility/initializer_list). I'll leave a comment!

File: tachyon/c/math/elliptic_curves/generator/ext_field.h.tpl [hunk 4]
```diff
@@ -33,19 +33,26 @@ struct tachyon_%{type}_fq%{degree} {
...
- * @brief Initializes an Fq%{degree} element to zero.
+ * @brief Returns a zero Fq%{degree} element.
  *
  * @return An Fq%{degree} element representing zero.
  */
 TACHYON_C_EXPORT tachyon_%{type}_fq%{degree} tachyon_%{type}_fq%{degree}_zero();
 
 /**
- * @brief Initializes an Fq%{degree} element to one.
+ * @brief Returns an one Fq%{degree} element.
```
Comment (TomTaehoonKim) [inline tachyon/c/math/elliptic_curves/generator/ext_field.h.tpl:43]:
```suggestion
 * @brief Returns a one Fq%{degree} element.
```

File: tachyon/math/finite_fields/generator/generator_util.cc (hunk 2 above)
Comment (TomTaehoonKim) [reply tachyon/math/finite_fields/generator/generator_util.cc:36]: Where is the initializer_list coming from?

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: tachyon/math/finite_fields/generator/generator_util.cc (hunk 2 above)
Comment (chokobole) [reply tachyon/math/finite_fields/generator/generator_util.cc:36]: Also in `bazel-bin/tachyon/math/finite_fields/test/gf7_config.h`:

```c++
  // Before, it was generated with UINT64_C()
  // constexpr static uint32_t kSubgroupGenerator = UINT64_C(5);
  // After, it is generated with a proper type.
  // constexpr static uint32_t kSubgroupGenerator = 5;
```

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by GideokKim — APPROVED** [review]
LGTM

**Review by batzor — APPROVED** [review]
LGTM

**Pull request merged**